### PR TITLE
Add a new html chat page with a hello message from server

### PR DESCRIPTION
**Pull request by ESaracay**
Week-3 chat implementation

### CODE REVIEW

**Review by zmr — APPROVED** [review]

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -43,7 +43,9 @@ function txtRotate(text) {
     text.deleting = false;
   }
   console.log(text.txt);
-  document.getElementById('intro_content').innerText = text.txt;
+  if(text.txt != null) {
```
Comment (tylerhou) [inline portfolio/src/main/webapp/script.js]: `if (`

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -89,3 +90,9 @@ randomMovie() {
 
   container.innerText = '"' + movie + '"';
 }
+
+async function grabName(){
+    const server = await fetch('/data');
+    const name = await server.text();
```
Comment (tylerhou) [inline portfolio/src/main/webapp/script.js]: Include a failure case?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -89,3 +47,22 @@ randomMovie() {
 
   container.innerText = '"' + movie + '"';
 }
+
+async function grabComments(){
+    const comments = await fetch('/data').then(response => response.json());
+    console.log(comments);
+    for (var i = 0; i < comments.length; i++) {
+        console.log("User:"+comments[i]["user"]);
```
Comment (tylerhou) [inline portfolio/src/main/webapp/script.js]: Please format this section; in particular there should be spaces around every `+`.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -89,3 +47,22 @@ randomMovie() {
 
   container.innerText = '"' + movie + '"';
 }
+
+async function grabComments(){
+    const comments = await fetch('/data').then(response => response.json());
+    console.log(comments);
```
Comment (tylerhou) [inline portfolio/src/main/webapp/script.js:56]: It's a good idea to remove console.logs so the end user doesn't see them.

If you really need debugging statements to stay around, I would define a function:

```js
function LOG() {
    if (false) {
        console.log.apply(console, arguments);
    }
}
```

Use it like `LOG("List of words: ", words)`.

Then in development you can change the hardcoded false to true.

File: walkthroughs/week-3-server/examples/random-quotes/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Converts a ServerStats instance into a JSON string using manual String concatentation.
+   */
+  private String convertToJson(ServerStats serverStats) {
```
Comment (tylerhou) [inline walkthroughs/week-3-server/examples/random-quotes/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java:50]: It's probably better to use Gson, but this works for now. JSON serialization is pretty tricky if you have to serialize objects with arbitrary strings.

I would add a TODO:

```
// TODO(esaracay): Use GSON to serialize.
String json = "{";
```

File: walkthroughs/week-3-server/examples/random-quotes/server-stats/src/main/webapp/script.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    statsListElement.appendChild(
+        createListElement('Current time: ' + stats.currentTime));
+    statsListElement.appendChild(
+        createListElement('Max memory: ' + stats.maxMemory));
+    statsListElement.appendChild(
+        createListElement('Used memory: ' + stats.usedMemory));
+  });
+}
+
+/** Creates an <li> element containing text. */
```
Comment (tylerhou) [inline walkthroughs/week-3-server/examples/random-quotes/server-stats/src/main/webapp/script.js:36]: Quality JSDocs.

**Review by tylerhou — CHANGES_REQUESTED** [review]

**Pull request merged**